plugins/catalog/src/components/EntityContextMenu/EntityContextMenu.tsx

```diff
@@ -142,37 +145,55 @@ export function EntityContextMenu(props: EntityContextMenuProps) {
         anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
         transformOrigin={{ vertical: 'top', horizontal: 'right' }}
         aria-labelledby="long-menu"
+        PaperProps={{
+          style: { minWidth: 200 },
+        }}
       >
         <MenuList autoFocusItem={Boolean(anchorEl)}>
           {extraItems}
-          <UnregisterEntity
-            unregisterEntityOptions={UNSTABLE_contextMenuOptions}
-            isUnregisterAllowed={isAllowed}
-            onUnregisterEntity={onUnregisterEntity}
-            onClose={onClose}
-          />
-          <MenuItem
-            onClick={() => {
-              onClose();
-              onInspectEntity();
-            }}
-          >
-            <ListItemIcon>
-              <BugReportIcon fontSize="small" />
-            </ListItemIcon>
-            <ListItemText primary={t('entityContextMenu.inspectMenuTitle')} />
-          </MenuItem>
-          <MenuItem
-            onClick={() => {
-              onClose();
-              copyToClipboard(window.location.toString());
-            }}
-          >
-            <ListItemIcon>
-              <FileCopyTwoToneIcon fontSize="small" />
-            </ListItemIcon>
-            <ListItemText primary={t('entityContextMenu.copyURLMenuTitle')} />
-          </MenuItem>
+          {contextMenuItems === undefined ? (
+            <>
+              <UnregisterEntity
+                unregisterEntityOptions={UNSTABLE_contextMenuOptions}
+                isUnregisterAllowed={isAllowed}
+                onUnregisterEntity={onUnregisterEntity}
+                onClose={onClose}
+              />
+              <MenuItem
+                onClick={() => {
+                  onClose();
+                  onInspectEntity();
+                }}
+              >
+                <ListItemIcon>
+                  <BugReportIcon fontSize="small" />
+                </ListItemIcon>
+                <ListItemText
+                  primary={t('entityContextMenu.inspectMenuTitle')}
+                />
+              </MenuItem>
+              <MenuItem
+                onClick={() => {
+                  onClose();
+                  copyToClipboard(window.location.toString());
+                }}
+              >
+                <ListItemIcon>
+                  <FileCopyTwoToneIcon fontSize="small" />
+                </ListItemIcon>
+                <ListItemText
+                  primary={t('entityContextMenu.copyURLMenuTitle')}
+                />
+              </MenuItem>
+            </>
```

Comment:
I believe you don't need to render the default items here because you are already providing them as default context menu item extensions.

Reply:
Ah, you're right, I thought we duplicated this component so we could introduce changes without having to worry about backwards compatibility with the legacy system. If I'm not mistaken, @Rugvip recommended me to do this when I was working on Entity groups and header extensions. Should we duplicate this one too @Rugvip?

Reply:
It depends, whatever is simplest without impacting the new or old design. In this case I think this is likely the way to go.